app/models/miq_request.rb

```diff
@@ -191,14 +191,18 @@ def must_have_user
   end
 
   def call_automate_event_queue(event_name)
-    MiqQueue.put(
+    q_options = {
       :class_name  => self.class.name,
       :instance_id => id,
       :method_name => "call_automate_event",
       :args        => [event_name],
       :zone        => options.fetch(:miq_zone, my_zone),
       :msg_timeout => 3600
-    )
+    }
+    user = User.current_user
```

Comment:
@lfu Who is going to be setting the User.current_user?
Is it the UI or the REST API?

Reply:
@mkanoor That varies depending on the events.

For ```request_created```, it seems by REST API.
For ```request_approved``` and ```request_starting```, it is by ```expose``` method.